OmgvaPOS/src/ReservationManagement/Service/ReservationService.cs

```diff
@@ -0,0 +1,60 @@
+using OmgvaPOS.Exceptions;
+using OmgvaPOS.ReservationManagement.DTOs;
+using OmgvaPOS.ReservationManagement.Mappers;
+using OmgvaPOS.ReservationManagement.Repository;
+using static OmgvaPOS.Exceptions.ExceptionMessages;
+
+namespace OmgvaPOS.ReservationManagement.Service
+{
+    public class ReservationService : IReservationService
+    {
+        private readonly IReservationRepository _repository;
+
+        public ReservationService(IReservationRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public IEnumerable<ReservationDto> GetAll()
+        {
+            var reservations = _repository.GetAll();
+            return ReservationMapper.ToDtoList(reservations);
+        }
+
+        public ReservationDto? GetById(long id)
+        {
+            var reservation = _repository.GetById(id);
+            return reservation == null ? null : ReservationMapper.ToDto(reservation);
+        }
+
+        public ReservationDto Create(CreateReservationDto createDto)
```

Comment:
It is important to validate if the reservation is created in the future and for the employee that is available.